src/features/counter/postsSlice.ts

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { getUserPosts } from '../../api/posts';
+import { Post } from '../../types/Post';
+
+export const loadUsersPosts
+  = createAsyncThunk('posts/fetch', (authorId: number) => {
+    return getUserPosts(authorId);
+  });
```

Comment:
just small improvement for better readability:
```suggestion
export const loadUsersPosts = createAsyncThunk(
  'posts/fetch',
  (authorId: number) => getUserPosts(authorId),
);
```